File: AoC2021/Day19/Day19.cs
using System;

/*

*/

namespace Day19
{
    class Program
    {
        private Program(string inputFile, bool part1)
        {
            var lines = AoC.Program.ReadLines(inputFile);

            if (part1)
            {
                var result1 = Part1(lines);
                Console.WriteLine($"Day19 : Result1 {result1}");
                var expected = 129;
                if (result1 != expected)
                {
                    throw new InvalidProgramException($"Part1 is broken {result1} != {expected}");
                }
            }
            else
            {
                var result2 = Part2(lines);
                Console.WriteLine($"Day19 : Result2 {result2}");
                var expected = 243;
                if (result2 != expected)
                {
                    throw new InvalidProgramException($"Part2 is broken {result2} != {expected}");
                }
            }
        }

        public static int Part1(string[] lines)
        {
            return 10;
        }

        public static int Part2(string[] lines)
        {
            return 20;
        }

        public static void Run()
        {
            Console.WriteLine("Day19 : Start");
            _ = new Program("Day19/input.txt", true);
            _ = new Program("Day19/input.txt", false);
            Console.WriteLine("Day19 : End");
        }
    }
}

File: AoC2021Swift/Day19.swift
import Foundation

/*
 
 
 */

class Day19
{
  func Execute(part1:Bool)
  {
    let lines = Program.ReadLines(day: "day19")
    Parse(lines:lines)

    if (part1)
    {
      let result = Part1()
      let expected = 2117664
      if (result != expected)
      {
        fatalError("Part1 is broken expected \(expected) got \(result)")
      }
      print("Day19 Part1: \(result)")
    }
    else
    {
      let result = Part2()
      let expected = 2073416724
      if (result != expected)
      {
        fatalError("Part2 is broken expected \(expected) got \(result)")
      }
      print("Day19 Part2: \(result)")
    }
  }
  
  func Parse(lines:[String]) {
    for l in lines {
      if l.isEmpty {
        continue
      }
    }
  }

  func Part1() -> Int {
    return 123
  }
  
  func Part2() -> Int {
    return 5443
  }
  
  func Run()
  {
    Execute(part1: true)
    Execute(part1: false)
  }
}
